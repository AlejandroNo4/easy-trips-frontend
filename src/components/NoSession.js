import React from 'react';
import PropTypes from 'prop-types';

const NoSession = ({ loginHandler, signUpHandler }) => {
  console.log('yay');
  return (
    <div className="bg-no-session d-flex justify-center align-center no-session-container">
      <div className="session-options d-flex space-between align-center flex-column">
        <h1 className="session-title">Not logged in</h1>
        <p>Please</p>
        <button onClick={loginHandler} type="button" className="session-btn white">
          Login
        </button>
        <p>or</p>
        <button onClick={signUpHandler} type="button" className="session-btn white">
          Sign Up
        </button>
      </div>
      <div className="bg-opacity" />
    </div>
  );
};

NoSession.defaultProps = {
  loginHandler: null,
  signUpHandler: null,
};

NoSession.propTypes = {
  loginHandler: PropTypes.func,
  signUpHandler: PropTypes.func,
};

export default NoSession;
