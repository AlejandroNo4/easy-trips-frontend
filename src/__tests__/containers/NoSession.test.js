import React from 'react';
import { render } from '@testing-library/react';
import NoSession from '../../components/NoSession';

describe('No Session component', () => {
  it('Render component wothout errors', () => {
    const component = render(<NoSession />);
    expect(component).toMatchSnapshot();
  });

  it('Precense of a button', () => {
    const component = render(<NoSession />);
    const button = component.getAllByRole('button');
    expect(button).toBeTruthy();
    expect(button.length).toBe(2);
    expect(button.length).not.toBe(0);
  });
});
