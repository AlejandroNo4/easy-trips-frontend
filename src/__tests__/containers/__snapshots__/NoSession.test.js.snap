// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`No Session component Render component wothout errors 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="bg-no-session d-flex justify-center align-center no-session-container"
      >
        <div
          class="session-options d-flex space-between align-center flex-column"
        >
          <h1
            class="session-title"
          >
            Not logged in
          </h1>
          <p>
            Please
          </p>
          <button
            class="session-btn white"
            type="button"
          >
            Login
          </button>
          <p>
            or
          </p>
          <button
            class="session-btn white"
            type="button"
          >
            Sign Up
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="bg-no-session d-flex justify-center align-center no-session-container"
    >
      <div
        class="session-options d-flex space-between align-center flex-column"
      >
        <h1
          class="session-title"
        >
          Not logged in
        </h1>
        <p>
          Please
        </p>
        <button
          class="session-btn white"
          type="button"
        >
          Login
        </button>
        <p>
          or
        </p>
        <button
          class="session-btn white"
          type="button"
        >
          Sign Up
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
